### DISCUSSION TITLE
Is it possible to publish a python package with mojo extensions, for example using setuptools?

### DISCUSSION THREAD

**Discussion by link89** [modular/modular#2553] [General]
I want to use mojo to speed up some critical algorithm in a python package but I get stuck at publishing the package. Is there any solution for publishing a python package with part of code implemented in mojo? I know there are setuptool support for C, C++ and rust, but I am not sure if there are similar solutions for mojo

Related discussions:
* https://github.com/modularml/mojo/discussions/546

**Comment by hogepodge** [comment]
We don't have a generic package management solution for Mojo. [While it's possible to build packages](https://docs.modular.com/mojo/manual/packages), we haven't yet built a publishing and distribution framework. We definitely recognize it as a much desired community feature, but unfortunately don't have a timeline that we can share for its development and release.

**Reply by link89** [reply]
I find that this is not a valid question as calling mojo from python is not supported yet.

**Reply by ml31415** [reply]
I guess that "calling mojo from python" was just die underlying question.
